main/options.c

```diff
@@ -2363,6 +2366,22 @@ static void processMaxRecursionDepthOption (const char *const option, const char
 	Option.maxRecursionDepth = atol(parameter);
 }
 
+static void processPatternLengthLimit(const char *const option, const char *const parameter)
+{
+	long patternLengthLimit;
+
+	if (parameter == NULL || parameter[0] == '\0')
+		error (FATAL, "A parameter is needed after \"%s\" option", option);
+
+	errno = 0;
+	patternLengthLimit = strtol(parameter, NULL, 0);
+
+	if (errno != 0 || patternLengthLimit < 0)
+		error (FATAL, "-%s: Invalid pattern length limit", option);
```

Comment:
@b4n Thanks for the hints. I guess it makes sense to add both `str_to_ulong` and `str_to_long` for future callers. Where should I put them?